### DISCUSSION TITLE
PlantUML Eclipse plugin not working in Eclipse 2023-09 ?

### DISCUSSION THREAD

**Discussion by JPG-IRIT** [hallvard/plantuml#175] [Q&A] [answered]
Hi,

  Is it possible something broke with the latest Eclipse release ? I can't seem to make it work anymore. I keep getting the following error message : 

![image](https://github.com/hallvard/plantuml/assets/113031570/95847780-3a04-4aa1-a8f1-7fdd1bfa2c3a)

  Am I doing something wrong ? Eclipse 2023-09 on Windows 10.
  
  Thanks !
  Jean-Pierre

**Comment by travkin79** [comment]
Hello @JPG-IRIT,
Could you please check your PlantUML preference main page? Do you see there that Graphviz is available like in the following screenshot? You error message looks to me as if the graphviz / DOT environment would be missing.

<img width="521" alt="image" src="https://github.com/hallvard/plantuml/assets/10887297/9ab87273-7683-4c18-810a-c003ebca2c4a">

**Comment by JPG-IRIT** [comment]
Thank you for your quick reply !

Yes, you are right : "no dot executable found" : 

![image](https://github.com/hallvard/plantuml/assets/113031570/81b7cfba-41c6-458a-8f7f-e6965077109a)
 
Weird, I never had that problem before using either the "Eclipse Marketplace" or "install new software" method (using this URL : http://hallvard.github.io/plantuml/).

**Comment by arnaudroques** [comment]
@JPG-IRIT Are you under Linux or Windows ?

**Reply by JPG-IRIT** [reply]
Windows 10

**Reply by arnaudroques** [reply]
Ok, thanks.

 _Theorically_, PlantUML distribution contains an [embedded, compiled version of GraphViz specifically for Windows](https://github.com/plantuml/plantuml/blob/master/BUILDING.md#-about-the-graphvizdat-file).
It is possible that PlantUML version 1.2023.10 has some issues with this embedded GraphViz (especially if we are talking about the EPL version). 
@travkin79 Could you check if there is a "graphviz.dat" file in the .jar archive (exact path is /net/sourceforge/plantuml/windowsdot/graphviz.dat")

**Reply by travkin79** [reply]
Hello @arnaudroques,
Thanks for looking at this issue. Indeed, there is no graphviz.dat in the EPL-based version of PlantUML 1.2023.10. I see only the file "WindowsDotArchive.class" in the path "/net/sourceforge/plantuml/windowsdot/".

**Reply by arnaudroques** [reply]
@travkin79 So I think that using PlantUML 1.2023.11 will fix this issue.
Is if possible to publish a new release with this version?

**Comment by travkin79** [comment]
Hi @arnaudroques,
Yes, I planned to release a new PlantUML Eclipse plug-in version anyway quite soon. I think, we could switch to PlantUML 1.2023.11 in that case. I'll check the EPL-based PlantUML vers. 1.2023.11, too, just to be sure.

**Reply by arnaudroques** [reply]
Yes, please check indeed EPL 1.2023.11 because I must say that I am getting lost with all those versions :-)

**Reply by travkin79** [reply]
Hi @arnaudroques, the EPL-based PlantUML 1.2023.11 contains the missing graphviz.dat file. So, upgrading the PlantUML eclipse plug-in to PlantUML 1.2023.11 library should hopefully fix the problem. I'll prepare a new PlantUML eclipse plug-in release.

@JPG-IRIT, could you test a new PlantUML Eclipse plug-in release candidate? I'm working on a Mac, so I cannot test that.

**Reply by JPG-IRIT** [reply]
Gladly, but I don't know where to find the file. Could you point me to it ?

**Reply by travkin79** [reply]
Hi @JPG-IRIT,
It would be enough to check, if the new PlantUML Eclipse plug-in release candidate works on your machine. :-) No need to check the files, @arnaudroques mentioned.

**Reply by JPG-IRIT** [reply]
Yes, but how do i install it : using "install new software" from URL (http://hallvard.github.io/plantuml) seems to only give me 1.1.29 or older. Is there another URL for release candidates or an option I have to check ? (sorry, I'm not that used to github)

**Reply by travkin79** [reply]
Hi @JPG-IRIT,
I'll build the new release candidate and will let you know how to install it from your side (I have to find out what is the easiest way for that).

**Reply by travkin79** [reply]
Hello @JPG-IRIT,
I've prepared a release candidate with the PlantUML Eclipse plug-in based on PlantUML library vers. 1.2023.11. Could you please download the given zip file, extract it on your machine, and install the release candidate using a local p2 repository (update site) from the extracted folder?

How to install:
1. Select Help -> Install new software.. -> Add... -> Local...
2. Choose the repository folder extracted from the given zip file -> name the update site, e.g. "PlantUML4Eclipse local" -> click the "Add" button
3. Install the two features
    - "PlantUML Ecore Feature" 1.1.30 and
    - "PlantUML Feature" 1.1.30


[Eclipse update site with PlantUML 4 eclipse release candidate (p2 repository): repository.zip](https://github.com/hallvard/plantuml/files/13024812/repository.zip)

The release candidate should automatically install the required "PlantUML Library Feature" 1.2023.11 and fix the described problem.

Please let me know if that works for you.

**Reply by JPG-IRIT** [reply]
It works perfectly, nicely done ! (tested on class diagrams with a 100 or so elements)
Thank you for looking into (and solving) this issue so fast.

On a related note : wouldn't it be possible to also update [the version on the Eclipse Marketplace](https://marketplace.eclipse.org/content/plantuml-plugin) ? I have no idea how it is linked to this repository and how it is done, but I think people often start by looking in this Marketplace.

**Reply by travkin79** [reply]
Great @JPG-IRIT,
I'll then prepare the official GitHub release with that fix.

Good point concerning the Eclipse Marketplace entry. Originally, this project was started by @hallvard. I think, he will have certain access rights to update the Marketplace entry for the PlantUML Eclipse plug-in. I didn't find any hint in this git repository on how to do that.

**Comment by travkin79** [comment, marked as answer]
Fixed in 1.1.30
